app/models/shipit/task.rb

```diff
@@ -333,5 +334,17 @@ def status_key
     def abort_key
       "#{status_key}:aborting"
     end
+
+    def log_output(text)
+      output_log_line_buffer.buffer(text) do |line|
+        Shipit.task_logger.info(
+          "[#{stack.repo_name}] #{line}"
```

Comment:
I’m not sure what's the best way to get the deploy id here. Is it safe to assume that if the task doesn't have a `deploy`, then it _is_ a deploy? i.e. `deploy ? deploy.id : id`?

Reply:
`Rollback < Deploy < Task`. So all `Deploy` are `Tasks` but the reverse isn't true.

But anyway, you don't really care wether it's a task or a deploy, in both case you want the record ID, so `id`.

Reply:
Thanks.